rundeckapp/grails-app/controllers/rundeck/interceptors/AuditInterceptor.groovy

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2017 Rundeck Inc, <http://rundeck.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package rundeck.interceptors
+
+import com.dtolabs.rundeck.core.audit.ActionTypes
+import com.dtolabs.rundeck.core.audit.ResourceTypes
+import com.dtolabs.rundeck.core.common.FrameworkResource
+import org.springframework.beans.factory.annotation.Autowired
+import rundeck.services.ConfigurationService
+
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * This filter tracks user access to trigger auditing events.
+ * For example, track the user access to different projects.
+ *
+ * This filter assumes {@link ProjectSelectInterceptor} has already processed the request.
+ *
+ * @author Alberto Hormazabal
+ */
+class AuditInterceptor {
```

Comment:
perhaps if no audit plugins are configured, this interceptor can short-circuit and do nothing?

Reply:
Nice idea.

Reply:
Done